### ISSUE TITLE
Allow the user to manually add webhooks

### ISSUE THREAD

**Issue by nsg** [canonical-websites/build.snapcraft.io#861]
I like to avoid granting different 3rd party systems unnecessary access to my GitHub account. A way to solve this would be to offer a manual setup where you ask the user to add the needed hooks manually.

This is of course not a option for most users so it's fine to hide the option in a corner or something :)

Over all, I have tested BSI now for a little over a day and nicely done! It was easy to setup to get a new snap published and I think I will move my other snaps to BSI as well.

**Comment by evandandrea** [comment]
Thanks! We'll be moving to daily builds by default, with an option to turn on builds for every git push.

Will that meet your needs?

**Comment by bartaz** [comment]
Also please note that currently BSI only has access to public repositories, because we don't ask for any permissions allowing us to know about user's private info.

**Comment by nsg** [comment]
For me, it is a question about security. Plan for the worst, what happens if BSI is hacked, or what can a evil person do (I'm sure you are all nice people)? My GitHub account has access to a large number of repositories that have nothing to do with BSI (or snaps) and I do not like the thought that BSI can do whatever it wants with these repos hooks.

![screenshot from 2017-08-17 13-45-48](https://user-images.githubusercontent.com/16388/29410775-6b0f0274-8352-11e7-96e5-cc46c11df919.png)

I think that's `admin:repo_hook` with the description of:
> Grants read, write, ping, and delete access to hooks in public or **private** repositories.

I do understand why you asks for this permission, it's really user friendly, and probably a sane default. But I would much prefer a manual setup alternative what do not asks for `admin:repo_hook`. It's not that much job to manually configure a webhook really. Maybe also combine this with the option to specify an alternative git url opening up the option to use BSI with self hosted solutions, Gitlab, Bitbucket and so on ...

Is this a big problem for me? No, not at all. I'm happily using BSI already but for the future the option to lock things down a litter better would be nice.

**Comment by nsg** [comment]
> Thanks! We'll be moving to daily builds by default, with an option to turn on builds for every git push.
> Will that meet your needs?

It do depends how it affects the permissions :)

Sounds good! I have probably wasted hundreds of builds when merging code to master, feels bad to waste resources. I assume you will only do a daily build if there has been a commit since the last build? I have a lot of projects that can be idle for weeks or months :) I also hope there will be a manual build option, can be useful on release day.

**Comment by bartaz** [comment]
We recently introduced manual builds and there is a work to introduce daily builds when dependencies on GH change (without commit to main repo).

We considered adding ability to add webhooks manually, but for now it seems to making the application flow more complicated for everyone to make it possible (as we can't ask for webhook permissions on sign in, even if users would be willing to give them).

Closing this for now.
